app/controllers/suggestions.rb

```diff
@@ -0,0 +1,73 @@
+get '/suggestions' do
+  @suggestion = Suggestion.all
+  @comment = Comment.all
+  erb :'suggestions/index'
+end
+
+get '/suggestions/new' do
+  @suggestion = Suggestion.new
+  erb :'suggestions/new'
+end
+
+post '/suggestions' do
+  @suggestion = Suggestion.new(params[:suggestion])
+  @suggestion.user = current_user
+  if @suggestion.save
+    redirect '/suggestions'
+  else
+    erb :'suggestions/new'
+  end
+end
+
+get '/suggestions/:id' do
+  @suggestion = Suggestion.find(params[:id])
+  erb :'suggestions/show'
+end
+
+get '/suggestions/:id/edit' do
+  @suggestion = Suggestion.find(params[:id])
+  if session[:user_id] == @suggestion.user
+    erb :'suggestions/edit'
+  else
+    erb :'suggestions/show'
+  end
+end
+
+patch '/suggestions/:id' do
+  @suggestion = Suggestion.find(params[:id])
+  @suggestion.update(params[:suggestion])
+
+  if @suggestion.save
+    redirect '/suggestions'
+  else
+    erb :'suggestions/edit'
+  end
+end
+
+delete '/suggestions/:id' do
+  @suggestion = Suggestion.find(params[:id])
+
+  @suggestion.destroy
+
+  redirect '/suggestions'
+end
+
+post '/suggestions/:id/upvote' do
+  @suggestion = Suggestion.find(params[:id])
+
+  @suggestion.update_attribute(:votes, @suggestion[:votes] + 1)
```

Comment:
You should have a separate vote table that creates records of votes rather than keeping track of a counter otherwise if a user unvotes you have no idea if they already did it.